### PR TITLE
New Selection API features

### PR DESCRIPTION
**Pull request by captainbrosset**
The Selection API has been evolving over the years, so I've spent some time here making the data match caniuse as best I could.
I think the support data makes sense.

I also really wanted to split out `getComposedRanges` because this is much newer, and Chromium is working on an implementation currently, and it unlocks a developer use case that was previously impossible.

### CODE REVIEW

File: features/selection-api-shadow.yml
Comment (captainbrosset) [inline features/selection-api-shadow.yml:1]: I don't like this feature ID at all. Let me know of better ideas if you have any.

File: features/selection-api.yml
Comment (captainbrosset) [inline features/selection-api.yml:1]: I couldn't name this just selection.yml, because that's already taken by the ::selection pseudo.

Comment (autonome) [reply features/selection-api.yml:1]: Matches both the spec title and the Caniuse key, so seems fine to me to keep `API` here, and there's precedent for doing so.

File: features/selection-api-shadow.yml
Comment (autonome) [reply features/selection-api-shadow.yml:1]: The primary aspect of feature seems to be in the API name itself, and the shadow tree boundary crossing is after the fact, as the MDN description notes:

> The Selection.getComposedRanges() method returns an array of [StaticRange](https://developer.mozilla.org/en-US/docs/Web/API/StaticRange) objects representing the current selection ranges, and can return ranges that potentially cross shadow boundaries.

Calling it by the API name in both the key and title might align more with what a developer would search for.

File: features/selection-api.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  - api.Selection.removeAllRanges
+  - api.Selection.selectAllChildren
+  - api.Selection.setBaseAndExtent
+  - api.Selection.setPosition
+  - api.Selection.toString
+  - api.Selection.type
+  - api.Window.getSelection
+
+  # The selectionchange events on <input> and <textarea> are not really needed.
+  # The event is fired on document too and his is what developers use.
```
Comment (jamesnw) [inline features/selection-api.yml:39]:
```suggestion
  # The event is fired on document too and this is what developers use.
```

File: features/selection-api-shadow.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+name: Selections across shadow tree boundaries
+description: The `window.getSelection().getComposedRanges()` method returns ranges that represent the current user selection, even if the selection spans across shadow tree boundaries.
+spec: https://w3c.github.io/selection-api/#dom-selection-getcomposedranges
+group: selection
```
Comment (jamesnw) [inline features/selection-api-shadow.yml:4]: Should this also go in a Shadow Dom group?

**Review by jamesnw — APPROVED** [review]
A few nits, but overall looks good!

File: features/selection-api-shadow.yml (hunk 2 above)
Comment (captainbrosset) [reply features/selection-api-shadow.yml:4]: There's no shadow dom group right now. There's a web components one though, so I'll add it to that one.

File: features/selection-api-shadow.yml
Comment (captainbrosset) [reply features/selection-api-shadow.yml:1]: Done, thanks.

**Pull request merged**